container/new_push.bzl

```diff
@@ -70,18 +75,40 @@ def _impl(ctx):
                 found = True
         if not found:
             fail("Did not find an index.json in the image attribute {} specified to {}".format(ctx.attr.image, ctx.label))
-    if ctx.attr.format == "legacy":
-        for f in ctx.files.image:
-            if f.basename == "manifest.json":
-                pusher_args += ["-src", "{index_dir}".format(
-                    index_dir = _get_runfile_path(ctx, f),
-                )]
     if ctx.attr.format == "docker":
         if len(ctx.files.image) == 0:
             fail("Attribute image {} to {} did not contain an image tarball".format(ctx.attr.image, ctx.label))
         if len(ctx.files.image) > 1:
             fail("Attribute image {} to {} had {} files. Expected exactly 1".format(ctx.attr.image, ctx.label, len(ctx.files.image)))
         pusher_args += ["-src", _get_runfile_path(ctx, ctx.files.image[0])]
+    if ctx.attr.format == "legacy":
+        # Construct container_parts for input to pusher.
+        image = _get_layers(ctx, ctx.label.name, ctx.attr.image)
+        blobsums = image.get("blobsum", [])
+        blobs = image.get("zipped_layer", [])
+        config = image["config"]
+        manifest = image["manifest"]
+        tarball = image.get("legacy")
+        image_files = blobs + blobsums
+        if tarball:
+            print("Pushing an image based on a tarball can be very " +
+                  "expensive.  If the image is the output of a " +
+                  "docker_build, consider dropping the '.tar' extension. " +
+                  "If the image is checked in, consider using " +
+                  "docker_import instead.")
+            image_files += [tarball]
+        if config:
+            image_files += [config]
+        if manifest:
+            image_files += [manifest]
+
+        legacy_dir = generate_legacy_dir(ctx, blobs, config)
```

Comment:
It seems you are not creating symlinks for all the files from image. Do you need to add those files to images_files and then in the runfiles?